### ISSUE TITLE
[sheets] ResourceNotFoundException

### ISSUE THREAD

**Issue by avently** [PocketByte/LocoLaser#19]
Hello. Tried to update my local strings from doc file but got an exception. Since I use the same config for many months I think the cause is maybe Google Docs API deprecation or something like that. Do you have the same error with your docs like me?

```                                                                             
Force import. All resource files will be refreshed.                                                                                                                              
Conflict strategy: remove_platform                                                                                                                                               
ERROR: java.lang.RuntimeException: com.google.gdata.util.ResourceNotFoundException: Not Found                                                                                    
<!DOCTYPE html><html lang="ru"><head><meta name="description" content="Веб-редактор текста, презентаций и таблиц"><meta name="viewport" content="width=device-width, initial-scal
e=1.0, maximum-scale=1.0, minimum-scale=1.0, user-scalable=0"><link rel="shortcut icon" href="//docs.google.com/favicon.ico"><title>Страница не найдена</title><meta name="referr
er" content="origin"><link href="//fonts.googleapis.com/css?family=Product+Sans" rel="stylesheet" type="text/css" nonce="ds"><style nonce="d
fd">/* Copyright 2021 Google Inc. All Rights Reserved. */   

..And so on...
```

**Comment by KamiSempai** [comment]
Please, try latest version 2.2.1

**Comment by avently** [comment]
@KamiSempai tried, working correctly. No excessive logs in terminal and good results of importing.
Thank you for fast fix!

**Comment by aleksey-ho** [comment]
I upgraded to 2.2.1 but still having the same exception

**Comment by avently** [comment]
@KamiSempai, same problem again. Are you able to reproduce?

**Comment by vvzoltan** [comment]
Same issue here as well. Tried upgrading to 2.2.1 but the issue remains.

**Comment by nectar1** [comment]
+1

**Comment by monr0e** [comment]
+1 same issue

**Comment by KamiSempai** [comment]
Fixed in version 2.2.2.
Publishing is in process. Artifacts should appear in a several hours. Publishing of plugin can take one or two days.

**Comment by avently** [comment]
@KamiSempai thanks! Can we avoid this problem in the future? Or it's unpredictable changes from Google?

**Comment by Gerz-inc** [comment]
> Fixed in version 2.2.2.
> Publishing is in process. Artifacts should appear in a several hours. Publishing of plugin can take one or two days.

That error fixed, but now I have

ERROR: java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248)
	at java.base/java.util.Objects.checkIndex(Objects.java:373)
	at java.base/java.util.ArrayList.get(ArrayList.java:427)
	at ru.pocketbyte.locolaser.google.sheet.GoogleSheet.fetchCellsIfNeeded(GoogleSheet.kt:262)
	at ru.pocketbyte.locolaser.google.sheet.GoogleSheet.getFirstRow(GoogleSheet.kt:210)
	at ru.pocketbyte.locolaser.resource.BaseTableResources.read(BaseTableResources.kt:76)
	at ru.pocketbyte.locolaser.LocoLaser.localizeInner(LocoLaser.kt:101)
	at ru.pocketbyte.locolaser.LocoLaser.localize(LocoLaser.kt:33)
	at ru.pocketbyte.locolaser.Main.main(Main.kt:21)

**Comment by KamiSempai** [comment]
@avently Google has turned off api v3. Now I did migration to api v4. In the next releases I will implement support of csv as a source to have an alternative for such cases.

**Comment by KamiSempai** [comment]
@Gerz-inc your issue fixed in version 2.2.3
